server/api/controllers/userSermonNotes/update-user-sermon-notes.js

```diff
@@ -0,0 +1,64 @@
+module.exports = {
+  friendlyName: 'Update user sermon note',
+
+  description: 'Update user sermon note',
+
+  inputs: {
+    sermonId: {
+      type: 'string',
+      required: true,
+    },
+    editedContent: {
+      type: 'string',
+    },
+    themes: {
+      type: 'string',
+    },
+    stickyNote: {
+      type: 'json',
+      defaultsTo: [],
+    },
+    isDeleted: { type: 'boolean' },
+  },
+
+  exits: {
+    success: {
+      description: 'User sermon note updated successfully',
+    },
+    invalid: {
+      description: 'Failed to update user sermon note',
+    },
+
+    missingRequiredFields: {
+      statusCode: 409,
+      description: 'Please fill in the required fields.',
+    },
+  },
+
+  fn: async function (
+    { sermonId, editedContent, themes, stickyNote, isDeleted },
+    exits
+  ) {
+    const user = this.req.user.fullName;
+    sails.log.info(`${user}: Updating user sermon note: ${sermonId}`);
+
+    if (sermonId) {
+      try {
+        let data = await UserSermonNotes.updateOne({
+          sermonId: sermonId,
```

Comment:
Don't do this, this is searching for all the sermon note with the same sermon, not the user's specific sermon note. You should add another params in the function to accept id, and pass it here to seach for the specific user's sermon note id in mongo
`_id:userSermonNoteId`

or better yet, you can do what you did in the get api, use the sermonId and userId to find the specific sermon note you want. Rmb updateOne only returns one object, not an array of objects like find().